### PR TITLE
Fat search api / google vision / video upload / image upload

### PR DESCRIPTION
**Pull request by calmoon808**
You can test the components in playground. Google vision only works in conjunction with fat search api, but fat search api can work without google vision. 

Video/Image upload components only puts the video/image url into store right now.

Also, if you're trying to test the new api I'm gonna need the IP address of the current network you're on that appears in the console.log from the actionsFatSecretFoodSearch response.

### CODE REVIEW

**Comment by calmoon808** [comment]
Also, if you're trying to test the new api I'm gonna need the IP address of the current network you're on that appears in the console.log from the actionsFatSecretFoodSearch response.

File: public/index.html
Comment (danieltichiyama) [inline public/index.html]: necessary?

File: server/api/exercise/tutorial_videos/index.js
Comment (danieltichiyama) [inline server/api/exercise/tutorial_videos/index.js]: necessary?

File: server/api/exercise/tutorial_videos/index.js
Comment (danieltichiyama) [inline server/api/exercise/tutorial_videos/index.js]: necessary?

File: server/api/nutrition/fat_secret/index.js
Comment (danieltichiyama) [inline server/api/nutrition/fat_secret/index.js:6]: could you please give these to us, so we can add it to our ENV files?

File: server/api/nutrition/fat_secret/index.js
Comment (danieltichiyama) [inline server/api/nutrition/fat_secret/index.js:64]: bomb ass code broh.

File: server/api/nutrition/fat_secret/index.js
Comment (danieltichiyama) [inline server/api/nutrition/fat_secret/index.js:81]: just wanna make sure, this shouldn't run every time we make a call to this route, right?  I will check when I try it out, but if you could confirm before I have a chance.

File: server/api/nutrition/nutrition/index.js
Comment (danieltichiyama) [inline server/api/nutrition/nutrition/index.js:71]: hahahaha

File: server/api/nutrition/vision/index.js
Comment (danieltichiyama) [inline server/api/nutrition/vision/index.js:20]: remove if unnecessary.

File: server/api/upload/image_upload/index.js
Comment (danieltichiyama) [inline server/api/upload/image_upload/index.js:10]: we need this in our ENV for it to work?

File: server/api/upload/video_upload/index.js
Comment (danieltichiyama) [inline server/api/upload/video_upload/index.js:10]: need this in our ENV?

File: server/database/decorator.js
Comment (danieltichiyama) [inline server/database/decorator.js:21]: rm if unnecessary

File: src/actions/index.js
Comment (danieltichiyama) [inline src/actions/index.js:133]: rm

File: src/actions/index.js
Comment (danieltichiyama) [inline src/actions/index.js:136]: rm

File: src/actions/index.js
Comment (danieltichiyama) [inline src/actions/index.js:372]: rm

File: src/actions/index.js
Comment (danieltichiyama) [inline src/actions/index.js:387]: rm?

File: src/components/FoodSearchComponent/FoodSearchComponent.jsx
Comment (danieltichiyama) [inline src/components/FoodSearchComponent/FoodSearchComponent.jsx:26]: rm?

File: src/components/FoodVisionComponent/FoodVisionComponent.jsx
Comment (danieltichiyama) [inline src/components/FoodVisionComponent/FoodVisionComponent.jsx]: rm?

File: src/reducers/index.js
Comment (danieltichiyama) [inline src/reducers/index.js]: rm?

File: src/reducers/index.js
Comment (danieltichiyama) [inline src/reducers/index.js]: rm?

File: src/reducers/index.js
Comment (danieltichiyama) [inline src/reducers/index.js]: rm?

File: src/components/LabelComponent/LabelComponent.jsx
Comment (danieltichiyama) [inline src/components/LabelComponent/LabelComponent.jsx:1]: sorry, not sure where this component gets used...?

**Review by danieltichiyama — CHANGES_REQUESTED** [review]

File: public/index.html
Comment (calmoon808) [reply public/index.html]: nope

File: server/api/exercise/tutorial_videos/index.js
Comment (calmoon808) [reply server/api/exercise/tutorial_videos/index.js]: nope

File: server/api/nutrition/fat_secret/index.js
Comment (calmoon808) [reply server/api/nutrition/fat_secret/index.js:6]: just posted to slack

File: server/api/nutrition/fat_secret/index.js
Comment (calmoon808) [reply server/api/nutrition/fat_secret/index.js:64]: ty

File: src/components/FoodVisionComponent/FoodVisionComponent.jsx
Comment (calmoon808) [reply src/components/FoodVisionComponent/FoodVisionComponent.jsx]: yep

File: src/reducers/index.js
Comment (calmoon808) [reply src/reducers/index.js]: yep

File: src/reducers/index.js
Comment (calmoon808) [reply src/reducers/index.js]: yep

File: src/reducers/index.js
Comment (calmoon808) [reply src/reducers/index.js]: yep

File: src/components/LabelComponent/LabelComponent.jsx
Comment (calmoon808) [reply src/components/LabelComponent/LabelComponent.jsx:1]: I'm still not 100% sure on what extending component does. I've been just putting that on all my components lol

File: server/api/exercise/tutorial_videos/index.js
Comment (calmoon808) [reply server/api/exercise/tutorial_videos/index.js]: nope

**Comment by calmoon808** [comment]
changes are made

**Review by danieltichiyama — COMMENTED** [review]
Okay, image upload works, just waiting on the API IP address approval to test the search function.  I'm thinking about how the image upload should function.  Is it possible to store the image until after they choose what food it is, and upon choosing to add the food to a meal, could we then prompt them to save the image then?

**Review by danieltichiyama — APPROVED** [review]

**Pull request merged**
